uwblueprint/ccbc
# Add Expired User Modal for Magazine Review Flow

## Notion ticket link

[Expired User for Magazine Review Flow](https://www.notion.so/uwblueprintexecs/Expired-User-for-Magazine-Review-Flow-55040038837b47888743fde175fc15ce?pvs=4)



## Implementation description
* Added a getUserByEmail function in the UsersAPIClient
* Used the Chakra UI Modal to block the user from accessing magazine review page 



## Steps to test
1. Open the home page 
2. A modal will popup blocking you from viewing the Magazine Review Page 
![image](https://user-images.githubusercontent.com/56615828/231363099-2ad1258e-252a-4dc3-9e9a-603105d3cd69.png)
3. Update your subscription expiry date to a date past the current date 
`docker exec -it ccbc-db-1 bash`
`psql -U postgres -d ccbc`
`UPDATE users SET subscription_expires_on='2023-07-25 01:22:25.607+00' WHERE id={your_user_db_id};`
4. Refresh the home page and you should be able to access the magazine review page
To set it back...
`UPDATE users SET subscription_expires_on='2023-03-31 01:22:25.607+00' WHERE id={your_user_db_id};`



## What should reviewers focus on?
* Styling and wording of the modal
* I removed the subscriptionExpiresOn body param that is sent from the frontend, since the backend already sets this for the User
* Added a useEffect hook to update the AuthenticatedUser state in the MagazineReview component


## Checklist
